docs/developer/README.md

```diff
@@ -0,0 +1 @@
+# Developer Documentation
```

Comment:
?

Reply:
this is "required" by the docs structure:
https://eclipse-edc.github.io/docs/#/submodule/GradlePlugins/docs/developer/

currently it's showing a 404 but it should at least show an empty page that it could be filled later on